### PR TITLE
Fix Update/Delete Category, Create/Delete/Get Product API

### PR DESCRIPTION
**Pull request by gamerpizza**

### CODE REVIEW

**Review by rogelioConsejo — APPROVED** [review]
It is hard to do a good Code Review because the system seems over-engineered and has clearly never been refactored. I understand some of the changes you are doing (like adding the URL to be able to fetch the image) and I am approving it on the basis that I trust that you tested it and it works. 
To give you an example, the "repository implementations" seems like a very unnecessarily complicated solution for what could be a simple `Persistence` interface with multiple implementations, and it seems like overkill to use generics to do something similar.
It is also very worrying that there are no unit tests. 
In short, it is dangerous to keep building on top of such a fragile and overly complicated system, and re-architecting it is both urgent and necessary. This is especially worrying since the system is not even in production and it is already unwieldy.
